### ISSUE TITLE
muscle_precice.cpp does not compile

### ISSUE THREAD

**Issue by maierbn** [maierbn/opendihu#44]
The example `opendihu/examples/electrophysiology/fibers/fibers_contraction/with_tendons_precice/muscle_precice.cpp` gives a compiler error: 
```
In file included from /workspace/opendihu/core/src/control/precice/surface_coupling/00_nested_solver.h:186,
                 from /workspace/opendihu/core/src/control/precice/surface_coupling/01_initialize.h:5,
                 from /workspace/opendihu/core/src/control/precice/surface_coupling/02_read_write.h:5,
                 from /workspace/opendihu/core/src/control/precice/surface_coupling/03_checkpoint.h:5,
                 from /workspace/opendihu/core/src/control/precice/surface_coupling/precice_adapter.h:6,
                 from /workspace/opendihu/core/src/opendihu.h:10,
                 from src/muscle_precice.cpp:8:
/workspace/opendihu/core/src/control/precice/surface_coupling/00_nested_solver.tpp: In instantiation of ‘void Control::PreciceAdapterNestedSolver<Control::Coupling<T1, MuscleContractionSolver<T2, T3> > >::saveFiberData(Control::PreciceAdapterNestedSolver<Control::Coupling<T1, MuscleContractionSolver<T2, T3> > >::NestedSolverType&) [with T1 = Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >; T2 = Mesh::StructuredDeformableOfDimension<3>; T3 = Equation::SolidMechanics::TransverselyIsotropicMooneyRivlinIncompressibleActive3D; Control::PreciceAdapterNestedSolver<Control::Coupling<T1, MuscleContractionSolver<T2, T3> > >::NestedSolverType = Control::Coupling<Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >, MuscleContractionSolver<> >]’:
/workspace/opendihu/core/src/control/precice/surface_coupling/precice_adapter.tpp:72:26:   required from ‘void Control::PreciceAdapter<NestedSolver>::run() [with NestedSolver = Control::Coupling<Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >, MuscleContractionSolver<> >]’
src/muscle_precice.cpp:40:14:   required from here
/workspace/opendihu/core/src/control/precice/surface_coupling/00_nested_solver.tpp:144:32: error: ‘class Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >’ has no member named ‘saveFiberDataCheckpoint’
  144 |   nestedSolver.timeStepping1().saveFiberDataCheckpoint();
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~
/workspace/opendihu/core/src/control/precice/surface_coupling/00_nested_solver.tpp: In instantiation of ‘void Control::PreciceAdapterNestedSolver<Control::Coupling<T1, MuscleContractionSolver<T2, T3> > >::loadFiberData(Control::PreciceAdapterNestedSolver<Control::Coupling<T1, MuscleContractionSolver<T2, T3> > >::NestedSolverType&) [with T1 = Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >; T2 = Mesh::StructuredDeformableOfDimension<3>; T3 = Equation::SolidMechanics::TransverselyIsotropicMooneyRivlinIncompressibleActive3D; Control::PreciceAdapterNestedSolver<Control::Coupling<T1, MuscleContractionSolver<T2, T3> > >::NestedSolverType = Control::Coupling<Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >, MuscleContractionSolver<> >]’:
/workspace/opendihu/core/src/control/precice/surface_coupling/precice_adapter.tpp:105:26:   required from ‘void Control::PreciceAdapter<NestedSolver>::run() [with NestedSolver = Control::Coupling<Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >, MuscleContractionSolver<> >]’
src/muscle_precice.cpp:40:14:   required from here
/workspace/opendihu/core/src/control/precice/surface_coupling/00_nested_solver.tpp:154:32: error: ‘class Control::MultipleInstances<OperatorSplitting::Godunov<Control::MultipleInstances<PrescribedValues<FunctionSpace::FunctionSpace<Mesh::StructuredDeformableOfDimension<1>, BasisFunction::LagrangeOfOrder<1> > > >, Dummy> >’ has no member named ‘restoreFiberDataCheckpoint’
  154 |   nestedSolver.timeStepping1().restoreFiberDataCheckpoint();
      |   ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~~~
scons: *** [build_release/src/muscle_precice.o] Error 1

```

**Comment by maierbn** [comment]
I disable it for now in `SConscript` to be able to build the docker container.

**Comment by carme-hp** [comment]
The error is now fixed. I added the corresponding empty method to the `MultipleInstances` class. I know this is not the ideal approach, but it's the best option I know without restructuring the adapter. 

As sanity check I tried running `opendihu/examples/electrophysiology/fibers/fibers_contraction/with_tendons_precice/multiple_tendons` for a few timesteps and no error was shown.

Note that I had to make minor changes to the `precice-config.xml` file to be able to run the case (e.g., the default network is for SuperMUC).
